### DISCUSSION TITLE
JavaScript intellisense

### DISCUSSION THREAD

**Discussion by anonimo82** [coder/code-server#4427] [Q&A] [answered]
Hello everyone.
I'm looking to get this: https://pasteboard.co/c0mEEXDVf8NA.png

But I don't know which extension I need.
Usually, I get fillRect and fillStyle, among other editor dependent suggestions (they're rarely the same from editor to editor) .
Screenshot above is from DreamWeaver.

Here is my code:
https://pastebin.com/EshzzGHm

I need this on Android device (I'm using Termux with code-server)
I tried various Android editors, no luck.

Thanks in advance for any help.

**Comment by jsjoeio** [comment]
According to VS Code's docs, they provide JavaScript intellisense out of the box. You shouldn't need to do anything. Do you have it turned off by chance?

![image](https://user-images.githubusercontent.com/3806031/139954358-6411add4-774a-428a-b9dd-41a8db55ed74.png)

Can you paste your `settings.json`?

**Comment by anonimo82** [comment]
In truth, I have code suggestions, but they're wrong.
The only working intellisense I had is the Dreamweaver's one, but I don't want to pay ~30 €/Month for it.

Here is a screenshot of the code completion I get from VSCode https://pasteboard.co/9UOfwWAHfaNj.png

Notice the differences with the Dreamweaver's one.

If you really need my settings.json, I'll upload it.

**Comment by anonimo82** [comment]
In VSCode, all works without any extension.
I discovered the IntelliSense technology is unable to get the type of canvas.getcontext("2d"); but, I made it to have IntelliSense working with Web Audio API and WebGL by downloading and editing examples.
This "bug" is common to all the editors I tried.
I guess we can mark this as answered now =)

**Reply by jsjoeio** [reply]
Nice debugging! Sorry the solution wasn't what you wanted but at least we know!

**Comment by anonimo82** [comment]
Uhm, no, I was wrong.
That kind of IntelliSense I had with DreamWeaver is not possible with any other editor I tried.

So, discussion is still open, is it there any good AddOn for this?
Something which works as Jock's SVG, but for JavaScript https://github.com/lishu/vscode-svg2

**Reply by jsjoeio** [reply]
Ah gotcha. Not that I know of unfortunately.

**Comment by anonimo82** [comment, marked as answer]
Someone solved this for me, look here: https://github.com/microsoft/vscode/issues/136717

**Reply by jsjoeio** [reply]
Hooray! That's awesome :D
